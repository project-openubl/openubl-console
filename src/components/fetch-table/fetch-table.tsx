import * as React from "react";
import {
  EmptyState,
  EmptyStateIcon,
  EmptyStateVariant,
  Title,
  EmptyStateBody,
} from "@patternfly/react-core";
import { ExclamationCircleIcon } from "@patternfly/react-icons";
import { global_danger_color_200 as globalDangerColor200 } from "@patternfly/react-tokens";
import {
  Table,
  TableHeader,
  TableBody,
  ICell,
  IRow,
  IActions,
} from "@patternfly/react-table";
import Skeleton from "@material-ui/lab/Skeleton";
import { FetchStatus } from "../../store/common";
import { Constants } from "../../Constants";

interface Props {
  columns: ICell[];
  rows: IRow[];
  actions: IActions;
  fetchStatus: FetchStatus;
  fetchError?: any;
}

export const FetchTable: React.FC<Props> = ({
  columns,
  rows,
  actions,
  fetchStatus,
  fetchError,
}) => {
  let rowsValue: IRow[] = rows;

  if (fetchStatus !== "complete") {
    rowsValue = [...Array(Constants.DEFAULT_PAGE_SIZE)].map(() => {
      return {
        cells: [...Array(columns.length)].map(() => ({
          title: <Skeleton />,
        })),
      };
    });
  }

  if (fetchError) {
    rowsValue = [
      {
        heightAuto: true,
        cells: [
          {
            props: { colSpan: columns.length },
            title: (
              <EmptyState variant={EmptyStateVariant.small}>
                <EmptyStateIcon
                  icon={ExclamationCircleIcon}
                  color={globalDangerColor200.value}
                />
                <Title headingLevel="h2" size="lg">
                  Unable to connect
                </Title>
                <EmptyStateBody>
                  There was an error retrieving data. Check your connection and
                  try again.
                </EmptyStateBody>
              </EmptyState>
            ),
          },
        ],
      },
    ];
  }

  return (
    <Table
      aria-label="Table"
      cells={columns}
      rows={rowsValue}
      actions={rowsValue === rows ? actions : undefined}
    >
      <TableHeader />
      <TableBody />
    </Table>
  );
};
