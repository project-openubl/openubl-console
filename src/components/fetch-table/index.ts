export { FetchTable as SimplePagination } from "./fetch-table";
