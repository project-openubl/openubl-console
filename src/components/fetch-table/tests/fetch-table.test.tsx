import * as React from "react";
import { shallow } from "enzyme";
import { FetchTable } from "../fetch-table";
import { ICell, IRow, IActions } from "@patternfly/react-table";

describe("FetchTable", () => {
  const columns: ICell[] = [
    { title: "Col1" },
    { title: "Col2" },
    { title: "Col3" },
  ];
  const rows: IRow[] = [...Array(15)].map((_, rowIndex) => {
    return {
      cells: [...Array(columns.length)].map((_, colIndex) => ({
        title: `${rowIndex},${colIndex}`,
      })),
    };
  });
  const actions: IActions = [
    {
      title: "Action1",
      onClick: jest.fn,
    },
    {
      title: "Action2",
      onClick: jest.fn,
    },
  ];

  it("Renders without crashing", () => {
    const wrapper = shallow(
      <FetchTable
        columns={columns}
        rows={rows}
        actions={actions}
        fetchStatus="complete"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
