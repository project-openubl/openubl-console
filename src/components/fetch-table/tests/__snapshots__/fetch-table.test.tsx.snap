// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FetchTable Renders without crashing 1`] = `
<Table
  actions={
    Array [
      Object {
        "onClick": [Function],
        "title": "Action1",
      },
      Object {
        "onClick": [Function],
        "title": "Action2",
      },
    ]
  }
  aria-label="Table"
  borders={true}
  canSelectAll={true}
  cells={
    Array [
      Object {
        "title": "Col1",
      },
      Object {
        "title": "Col2",
      },
      Object {
        "title": "Col3",
      },
    ]
  }
  className=""
  contentId="expanded-content"
  dropdownDirection="down"
  dropdownPosition="right"
  expandId="expandable-toggle"
  gridBreakPoint="grid-md"
  role="grid"
  rowLabeledBy="simple-node"
  rows={
    Array [
      Object {
        "cells": Array [
          Object {
            "title": "0,0",
          },
          Object {
            "title": "0,1",
          },
          Object {
            "title": "0,2",
          },
        ],
      },
      Object {
        "cells": Array [
          Object {
            "title": "1,0",
          },
          Object {
            "title": "1,1",
          },
          Object {
            "title": "1,2",
          },
        ],
      },
      Object {
        "cells": Array [
          Object {
            "title": "2,0",
          },
          Object {
            "title": "2,1",
          },
          Object {
            "title": "2,2",
          },
        ],
      },
      Object {
        "cells": Array [
          Object {
            "title": "3,0",
          },
          Object {
            "title": "3,1",
          },
          Object {
            "title": "3,2",
          },
        ],
      },
      Object {
        "cells": Array [
          Object {
            "title": "4,0",
          },
          Object {
            "title": "4,1",
          },
          Object {
            "title": "4,2",
          },
        ],
      },
      Object {
        "cells": Array [
          Object {
            "title": "5,0",
          },
          Object {
            "title": "5,1",
          },
          Object {
            "title": "5,2",
          },
        ],
      },
      Object {
        "cells": Array [
          Object {
            "title": "6,0",
          },
          Object {
            "title": "6,1",
          },
          Object {
            "title": "6,2",
          },
        ],
      },
      Object {
        "cells": Array [
          Object {
            "title": "7,0",
          },
          Object {
            "title": "7,1",
          },
          Object {
            "title": "7,2",
          },
        ],
      },
      Object {
        "cells": Array [
          Object {
            "title": "8,0",
          },
          Object {
            "title": "8,1",
          },
          Object {
            "title": "8,2",
          },
        ],
      },
      Object {
        "cells": Array [
          Object {
            "title": "9,0",
          },
          Object {
            "title": "9,1",
          },
          Object {
            "title": "9,2",
          },
        ],
      },
      Object {
        "cells": Array [
          Object {
            "title": "10,0",
          },
          Object {
            "title": "10,1",
          },
          Object {
            "title": "10,2",
          },
        ],
      },
      Object {
        "cells": Array [
          Object {
            "title": "11,0",
          },
          Object {
            "title": "11,1",
          },
          Object {
            "title": "11,2",
          },
        ],
      },
      Object {
        "cells": Array [
          Object {
            "title": "12,0",
          },
          Object {
            "title": "12,1",
          },
          Object {
            "title": "12,2",
          },
        ],
      },
      Object {
        "cells": Array [
          Object {
            "title": "13,0",
          },
          Object {
            "title": "13,1",
          },
          Object {
            "title": "13,2",
          },
        ],
      },
      Object {
        "cells": Array [
          Object {
            "title": "14,0",
          },
          Object {
            "title": "14,1",
          },
          Object {
            "title": "14,2",
          },
        ],
      },
    ]
  }
  variant={null}
>
  <TableHeader />
  <Component />
</Table>
`;
